commands/audit/jasrunner.go

```diff
@@ -2,62 +2,119 @@ package audit
 
 import (
 	"errors"
+	"fmt"
+	"github.com/jfrog/gofrog/parallel"
+	jfrogappsconfig "github.com/jfrog/jfrog-apps-config/go"
 	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
 	"github.com/jfrog/jfrog-cli-security/commands/audit/jas"
 	"github.com/jfrog/jfrog-cli-security/commands/audit/jas/applicability"
 	"github.com/jfrog/jfrog-cli-security/commands/audit/jas/iac"
 	"github.com/jfrog/jfrog-cli-security/commands/audit/jas/sast"
 	"github.com/jfrog/jfrog-cli-security/commands/audit/jas/secrets"
 	"github.com/jfrog/jfrog-cli-security/utils"
-	"github.com/jfrog/jfrog-client-go/utils/io"
 	"github.com/jfrog/jfrog-client-go/utils/log"
 )
 
-func runJasScannersAndSetResults(scanResults *utils.Results, directDependencies []string,
-	serverDetails *config.ServerDetails, workingDirs []string, progress io.ProgressMgr, thirdPartyApplicabilityScan bool, msi string) (err error) {
+func RunJasScannersAndSetResults(auditParallelRunner *utils.AuditParallelRunner, scanResults *utils.Results, serverDetails *config.ServerDetails,
+	auditParams *AuditParams, jfrogAppsConfig *jfrogappsconfig.JFrogAppsConfig, msi string) (err error) {
 	if serverDetails == nil || len(serverDetails.Url) == 0 {
 		log.Warn("To include 'Advanced Security' scan as part of the audit output, please run the 'jf c add' command before running this command.")
 		return
 	}
-	scanner, err := jas.NewJasScanner(workingDirs, serverDetails)
+	scanner, err := jas.NewJasScanner(serverDetails, jfrogAppsConfig)
 	if err != nil {
 		return
 	}
+	// Set environments variables for analytics in analyzers manager.
+	callback := jas.SetAnalyticsMetricsDataForAnalyzerManager(msi, scanResults.GetScaScannedTechnologies())
 	defer func() {
+		auditParallelRunner.ScannersWg.Wait()
+		callback()
 		cleanup := scanner.ScannerDirCleanupFunc
 		err = errors.Join(err, cleanup())
 	}()
-	if progress != nil {
-		progress.SetHeadlineMsg("Running applicability scanning")
-	}
-	// Set environments variables for analytics in analyzers manager.
-	callback := jas.SetAnalyticsMetricsDataForAnalyzerManager(msi, scanResults.GetScaScannedTechnologies())
-	defer callback()
-	scanResults.ExtendedScanResults.ApplicabilityScanResults, err = applicability.RunApplicabilityScan(scanResults.GetScaScansXrayResults(), directDependencies, scanResults.GetScaScannedTechnologies(), scanner, thirdPartyApplicabilityScan)
-	if err != nil {
-		return
-	}
+
 	// Don't execute other scanners when scanning third party dependencies.
-	if thirdPartyApplicabilityScan {
-		return
+	if !auditParams.thirdPartyApplicabilityScan {
+		for _, module := range scanner.JFrogAppsConfig.Modules {
```

Comment:
```suggestion
	if !auditParams.thirdPartyApplicabilityScan {
        err := addJasScanTasks(...)
```
I think extracting the content inside the if statment will help readability